spec/support/json_manager.rb

```diff
@@ -5,7 +5,12 @@ class JsonManager
   field :name
   field :ssn, default: '123-45-6789'
   has_many :json_employees
-  belongs_to :supervisor, class_name: 'JsonSupervisor'
+
+  if Mongoid::VERSION < "6.0.0"
```

Comment:
This is not a valid comparison! Use https://github.com/mongoid/mongoid-compatibility 's `Mongoid::Compatibility::Version.mongoid6_or_older?`

Reply:
It is totally a valid comparison that people think is silly but works totally fine:

```ruby
"6.0.1" < "6.0.0" # => false
"6.0.0" < "6.0.0" # => false
"5.10.10" < "6.0.0" # => true
"5.9.9" < "6.0.0" # => true
```

I don't think we need a dependency.

Reply:
Let's take a dependency?

```ruby
"6.0.0" > "6.0"
# => true
```

Reply:
Also note that the dependency would only be for tests.

Reply:
Another example that fails.

```ruby
"6.0.0" < "6.0.0.beta" 
# => true
```

Reply:
I'd use `Gem::Version` if a more accurate comparison is necessary: 

```ruby
Gem::Version.new("6.0.0") < Gem::Version.new("6.0") # => false
Gem::Version.new("6.0.1") < Gem::Version.new("6.0.0") # => false
Gem::Version.new("6.0.0") < Gem::Version.new("6.0.0") # => false
Gem::Version.new("5.10.10") < Gem::Version.new("6.0.0") # => true
Gem::Version.new("5.9.9") < Gem::Version.new("6.0.0") # => true
```

Reply:
That's what mongoid-compatibility does, but it's future-proof.

https://github.com/mongoid/mongoid-compatibility/blob/master/lib/mongoid/compatibility/version.rb

Reply:
> as it checks major version only and has semantic verbs like mongoid6_or_older? which are easier to read.

This doesn't sound like a strong counter objection when the `Gem::Version` comparizon is very simple and easy to understand. Is this something that might take more than an hour to understand, or something that really requires encapsulation?

> since we were doing these comparisons in 10 different ways all over the place.

Here we are not trying to understand 10 different sorting algorithms, but just a version comparison, basically a number comparison internally in `Gem::Version`. Also, could you actually really list 10 different examples? I have seen a few, but I'm sure it doesn't exceed 10.

Reply:
I noticed it's already a dependency of this project, so I just switched it anyway. However, I'm not convinced that this is a better way as `Gem::Version` is a more generic abstraction that can be used almost anywhere else, and it doesn't require any specific knowledge about Mongoid.

Reply:
I think it's all a small deal and either works!